### PR TITLE
 Refactor controller to use faas-netes FunctionFactory

### PR DESCRIPTION
**Pull request by stefanprodan**
## Description

Use faas-netes k8s package for health-check probes, read-only root filesystem and non root user ID.

- allow health-check custom path and delay as in https://github.com/openfaas/faas-netes/pull/447
- remove deployment and service informer handlers since reconciliation uses annotations to detect  changes
- fix false-positive errors during reconciliation (reset err after not found type is handled)

## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

## How Has This Been Tested?

Tested locally with Kubernetes Kind.

Clone faas-netes and run `make start-kind`. Deploy the operator with:

```
helm  upgrade -i openfaas ./chart/openfaas \
    --namespace openfaas  \
    --set basic_auth=true \
    --set functionNamespace=openfaas-fn \
    --set operator.create=true \
    --set operator.image=stefanprodan/openfaas-operator:factory-v2 \
    --wait
```

Deploy kubesec from this branch:

```
kubectl apply -f artifacts/kubesec.yaml
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by stefanprodan** [comment]
@alexellis @LucasRoesler this is ready for review now

File: main.go [hunk 1]
```diff
@@ -90,14 +110,14 @@ func main() {
...
-		config,
+		factory,
 	)
 
 	go kubeInformerFactory.Start(stopCh)
 	go faasInformerFactory.Start(stopCh)
 	go server.Start(faasClient, kubeClient, kubeInformerFactory)
 
-	if err = ctrl.Run(2, stopCh); err != nil {
+	if err = ctrl.Run(1, stopCh); err != nil {
```
Comment (alexellis) [inline main.go]: what does this change mean? 2->1?

**Comment by alexellis** [comment]
Can you tell me about the DeploymentInformer / ServiceInformer going away?

**Comment by stefanprodan** [comment]
We don't need to react to deployments or services events since the reconciliation loop will detect any external change and will roll it back.

**Pull request merged**